src/app/payments/translations.ts

```diff
@@ -0,0 +1,73 @@
+import { LedgerTransactionType } from "@domain/ledger"
+import {
+  MissingPropsInTransactionForPaymentFlowError,
+  NonLnPaymentTransactionForPaymentFlowError,
+  PaymentFlow,
+} from "@domain/payments"
+import {
+  paymentAmountFromCents,
+  paymentAmountFromSats,
+  WalletCurrency,
+} from "@domain/shared"
+import { PaymentInitiationMethod, SettlementMethod } from "@domain/wallets"
+
+export const PaymentFlowFromLedgerTransaction = <
+  S extends WalletCurrency,
+  R extends WalletCurrency,
+>(
+  ledgerTxn: LedgerTransaction<S>,
+): PaymentFlow<S, R> | ValidationError => {
+  if (ledgerTxn.type !== LedgerTransactionType.Payment) {
+    return new NonLnPaymentTransactionForPaymentFlowError()
```

Comment:
How are on-chain payments updated?

Reply:
In trigger on each new block I believe: https://github.com/GaloyMoney/galoy/blob/fc621181ade51e539b87cb41132a70375620dd5d/src/servers/trigger.ts#L119